lib/ex_doc/autolink.ex

```diff
@@ -0,0 +1,397 @@
+         [[_, text]] <- Regex.scan(~r/^`(.+)`$/, href) do
+      case text_to_ref(text, :custom_link, config) do
+        :no_ref ->
+          nil
+
+        ref ->
+          url(ref, config)
+      end
+    else
+      _ -> nil
+    end
+  end
+
+  defp text_to_ref("c:" <> text, mode, config) do
+    with {call, {:function, module, name, arity}} <- text_to_ref(text, mode, config) do
+      {call, {:callback, module, name, arity}}
+    end
+  end
+
+  defp text_to_ref("t:" <> text, mode, config) do
+    with {call, {:function, module, name, arity}} <- text_to_ref(text, mode, config) do
+      {call, {:type, module, name, arity}}
+    end
+  end
+
+  defp text_to_ref("mix help " <> name, _mode, _config), do: mix_task(name)
+  defp text_to_ref("mix " <> name, _mode, _config), do: mix_task(name)
+
+  defp text_to_ref(text, mode, config) do
+    if not String.contains?(text, [" ", "(", ")"]) do
+      case Code.string_to_quoted(text) do
+        {:ok, {:__aliases__, _, _} = module} ->
+          {:module, module(module)}
+
+        {:ok, {:/, _, [{{:., _, [module, name]}, _, []}, arity]}}
+        when is_atom(name) and is_integer(arity) ->
+          {:remote, {:function, module(module), name, arity}}
+
+        {:ok, {:/, _, [{name, _, _}, arity]}}
+        when is_atom(name) and is_integer(arity) ->
+          {:local, {:function, config.current_module, name, arity}}
+
+        {:ok, erlang_module} when is_atom(erlang_module) and mode == :custom_link ->
+          {:module, erlang_module}
+
+        _ ->
+          :no_ref
+      end
+    else
+      :no_ref
+    end
+  end
+
+  defp module({:__aliases__, _, _} = module), do: Module.concat([Macro.to_string(module)])
+  defp module(module) when is_atom(module), do: module
+
+  defp mix_task(name) do
+    if name =~ ~r/^[a-z][a-z0-9]*(\.[a-z][a-z0-9]*)*$/ do
+      parts = name |> String.split(".") |> Enum.map(&Macro.camelize/1)
+      {:module, Module.concat([Mix, Tasks | parts])}
+    else
+      :no_ref
+    end
+  end
+
+  @basic_types [
+    any: 0,
+    none: 0,
+    atom: 0,
+    map: 0,
+    pid: 0,
+    port: 0,
+    reference: 0,
+    struct: 0,
+    tuple: 0,
+    integer: 0,
+    float: 0,
+    neg_integer: 0,
+    non_neg_integer: 0,
+    pos_integer: 0,
+    list: 1,
+    nonempty_list: 1,
+    improper_list: 2,
+    maybe_improper_list: 2
+  ]
+
+  @built_in_types [
+    term: 0,
+    arity: 0,
+    as_boolean: 1,
+    binary: 0,
+    bitstring: 0,
+    boolean: 0,
+    byte: 0,
+    char: 0,
+    charlist: 0,
+    nonempty_charlist: 0,
+    fun: 0,
+    function: 0,
+    identifier: 0,
+    iodata: 0,
+    iolist: 0,
+    keyword: 0,
+    keyword: 1,
+    list: 0,
+    nonempty_list: 0,
+    maybe_improper_list: 0,
+    nonempty_maybe_improper_list: 0,
+    mfa: 0,
+    module: 0,
+    no_return: 0,
+    node: 0,
+    number: 0,
+    struct: 0,
+    timeout: 0
+  ]
+
+  @doc """
+  Converts given types/specs `ast` into HTML with links.
+  """
+  def typespec(ast, options) do
```

Comment:
this is pretty much unchanged from what we currently have on master. There's an AST-based approach that we could use but it's out of the scope of this PR, we'll go back to it when we support Erlang. Here's the gist: https://gist.github.com/wojtekmach/049ce0c90e4e7ee1da05491b69d7ac62